WordPress/Classes/Services/BlogJetpackSettingsService.swift

```diff
@@ -79,51 +79,51 @@ struct BlogJetpackSettingsService {
             return
         }
         guard let remoteAPI = blog.wordPressComRestApi(),
-            let blogDotComId = blog.dotComID as? Int,
-            let blogSettings = blog.settings else {
-                failure(nil)
-                return
+            let blogDotComId = blog.dotComID as? Int else {
+            failure(nil)
+            return
         }
 
         let remote = BlogJetpackSettingsServiceRemote(wordPressComRestApi: remoteAPI)
-        remote.getJetpackModulesSettingsForSite(blogDotComId,
-                                                success: { (remoteModulesSettings) in
-                                                    self.updateJetpackModulesSettings(blogSettings, remoteSettings: remoteModulesSettings)
-                                                    do {
-                                                        try self.context.save()
-                                                        success()
-                                                    } catch let error as NSError {
-                                                        failure(error)
-                                                    }
-                                                },
-                                                failure: { (error) in
-                                                    failure(error)
-                                                })
+        remote.getJetpackModulesSettingsForSite(
+            blogDotComId,
+            success: { (remoteModulesSettings) in
+                self.coreDataStack.performAndSave({ context in
+                    guard let blogSettings = Blog.lookup(withObjectID: blog.objectID, in: context)?.settings else {
+                        return
+                    }
+                    self.updateJetpackModulesSettings(blogSettings, remoteSettings: remoteModulesSettings)
+                }, completion: success)
+            },
+            failure: failure
+        )
     }
 
     func updateJetpackSettingsForBlog(_ blog: Blog, success: @escaping () -> Void, failure: @escaping (Error?) -> Void) {
         guard let remoteAPI = blog.wordPressComRestApi(),
             let blogDotComId = blog.dotComID as? Int,
             let blogSettings = blog.settings else {
-                failure(nil)
-                return
+            failure(nil)
+            return
```

Comment:
This indentation seems worse IMHO.

Maybe the `else {` should be on a new line?

Reply:
The original indentation looked weird to me and I went with the one that Xcode suggested. Updated in https://github.com/wordpress-mobile/WordPress-iOS/pull/19920/commits/bd5570c05240b961f595c3cb9f9f4b4e93c2ae2c